src/compiler/program.ts

```diff
@@ -4549,8 +4564,8 @@ export function createProgram(rootNamesOrOptions: readonly string[] | CreateProg
         if (!symlinks) {
             symlinks = createSymlinkCache(currentDirectory, getCanonicalFileName);
         }
-        if (files && resolvedTypeReferenceDirectives && !symlinks.hasProcessedResolutions()) {
-            symlinks.setSymlinksFromResolutions(files, resolvedTypeReferenceDirectives);
+        if (files && automaticTypeDirectiveResolutions && !symlinks.hasProcessedResolutions()) {
+            symlinks.setSymlinksFromResolutions(files, automaticTypeDirectiveResolutions);

```

Comment:
Why this change? If someone sets an empty `types` array but uses a bunch of triple slash type reference directives that resolve into symlinked node_modules, we’ll miss those symlinks.

Reply:
we are iterating over file.resolvedTypeReferenceDirectives so that should get covered?